pkg/util/gce/gce.go

```diff
@@ -83,6 +83,19 @@ func GetClusterName() (string, error) {
 	return clusterName, nil
 }
 
+// GetPublicIPv4 returns the public IPv4 address of the current GCE instance
+func GetPublicIPv4() (string, error) {
+	if !config.IsCloudProviderEnabled(CloudProviderName) {
+		return "", fmt.Errorf("cloud provider is disabled by configuration")
+	}
+	clusterName, err := getResponseWithMaxLength(metadataURL+"/instance/network-interfaces/0/access-configs/0/external-ip",
+		config.Datadog.GetInt("metadata_endpoints_max_hostname_size"))
+	if err != nil {
+		return "", fmt.Errorf("unable to retrieve public IPv4 from GCE: %s", err)
+	}
+	return clusterName, nil
+}
```

Comment:
I think we can directly return here too and print the warning in the metadata (like for ec2).

```suggestion
	return getResponseWithMaxLength(metadataURL+"/instance/network-interfaces/0/access-configs/0/external-ip",
		config.Datadog.GetInt("metadata_endpoints_max_hostname_size"))
}
```

Reply:
This copies the pattern of the other Get* functions, and I think adds a bit of context to the error. However, I do still have "clusterName" from copy-pasting, so I'll fix that :)